CMakeLists.txt

```diff
@@ -46,6 +46,7 @@ option(ENABLE_PROGRAMS "Build mbed TLS programs." ON)
 
 option(UNSAFE_BUILD "Allow unsafe builds. These builds ARE NOT SECURE." OFF)
 option(MBEDTLS_FATAL_WARNINGS "Compiler warnings treated as errors" ON)
+option(DEV_MODE "Development mode: (re)generate some files as needed" ON)
```

Comment:
“Development mode” is rather vague: it could refer to debug builds, for example. But I don't have a better name to suggest.

Reply:
I'm not too fond of that name either. Let's hope the second reviewer will have better ideas than us.

Reply:
Is there a problem with something like `GEN_FILES` / "Generate the auto-generated files as needed"?

Reply:
Nope, that looks like a better name and description indeed.